extensions/markdown-language-features/src/test/pathCompletion.test.ts

```diff
@@ -171,4 +171,54 @@ suite('Markdown path completion provider', () => {
 
 		assert.ok(completions.some(x => x.insertText === 'file.md'), 'Has file from space');
 	});
+
+	test('Should complete path for angled brackets', async () => {
+		const completions = await getCompletionsAtCursor(workspacePath('new.md'), joinLines(
+			`[](<${CURSOR}>)`
+		));
+
+		assert.ok(completions.some(x => x.label === 'a.md'), 'Has a.md file completion');
+		assert.ok(completions.some(x => x.label === 'b.md'), 'Has b.md file completion');
+		assert.ok(completions.some(x => x.label === 'sub/'), 'Has sub folder completion');
+	});
+
+	test('Should complete path for incomplete links', async () => {
+		const completions = await getCompletionsAtCursor(workspacePath('new.md'), joinLines(
+			`[](<${CURSOR}`
+		));
+
+		assert.ok(completions.some(x => x.label === 'a.md'), 'Has a.md file completion');
+		assert.ok(completions.some(x => x.label === 'b.md'), 'Has b.md file completion');
+		assert.ok(completions.some(x => x.label === 'sub/'), 'Has sub folder completion');
+	});
+
+	test('Should complete links that are on the line', async () => {
+		const completions = await getCompletionsAtCursor(workspacePath('new.md'), joinLines(
+			`text [](<${CURSOR}>) more text`
+		));
+
+		assert.ok(completions.some(x => x.label === 'a.md'), 'Has a.md file completion');
+		assert.ok(completions.some(x => x.label === 'b.md'), 'Has b.md file completion');
+		assert.ok(completions.some(x => x.label === 'sub/'), 'Has sub folder completion');
+	});
+
+	test('Should complete links for links with titles', async () => {
+		const completions = await getCompletionsAtCursor(workspacePath('new.md'), joinLines(
+			`[](<${CURSOR}> "title">`
+		));
+
+		assert.ok(completions.some(x => x.label === 'a.md'), 'Has a.md file completion');
+		assert.ok(completions.some(x => x.label === 'b.md'), 'Has b.md file completion');
+		assert.ok(completions.some(x => x.label === 'sub/'), 'Has sub folder completion');
+	});
+
+	test('Path component should not be encoded with <>', async () => {
+		const completions = await getCompletionsAtCursor(workspacePath('new.md'), joinLines(
```

Comment:
This doesn't actually test that encoding isn't happening. For that you need to check the completions for a path with spaces in it. See `Should escape spaces in path names` as an example

Reply:
This also isn't testing the new behavior. This should test that completions inside `<>` links don't encode paths with spaces in them

Reply:
Sorry but this is still incorrect (and I believe will also fail as written at the moment). Here's what the test needs to be:

For any of the cases:

```
[](<./CURSOR
or 
[](</CURSOR
or 
[](<CURSOR
```

Completions for paths with spaces in their name use an  insert text that does not `%20` encode the spaces:

```
assert.ok(completions.some(x => x.insertText === 'sub with space/'), 'Has sub folder completion');
```